src/pal/src/init/pal.cpp

```diff
@@ -1150,269 +1146,61 @@ Notes 2:
-
-        /* path_ptr is NULL if there's no ':' after this directory */
-        cur_dir=path_ptr;
+        ERROR("realpath() failed!\n");
+        return NULL;
     }
 
-    free(env_path);
-    TRACE("No %s found in $PATH (%s)\n", exe_name, EnvironGetenv("PATH", FALSE));
+    return_size = MultiByteToWideChar(CP_ACP,0,real_path,-1,NULL,0);
+    if (0 == return_size)
+    {
+        ASSERT("MultiByteToWideChar failure\n");
+        return NULL;
+    }
 
-last_resort:
-    /* last resort : see if the executable is in the current directory. This is
-       possible if it comes from a exec*() call. */
-    if (0 == stat(exe_name,&theStats))
+    return_value = reinterpret_cast<LPWSTR>(InternalMalloc((return_size*sizeof(WCHAR))));
+    if (NULL == return_value)
+    {
+        ERROR("Not enough memory to create full path\n");
+        return NULL;
+    }
+    else
     {
-        if ( UTIL_IsExecuteBitsSet( &theStats ) )
+        if (!MultiByteToWideChar(CP_ACP, 0, real_path, -1,
+                                return_value, return_size))
         {
-            if (!CorUnix::RealPathHelper(exe_name, real_path))
-            {
-                ERROR("realpath() failed!\n");
-                return NULL;
-            }
-
-            return_size = MultiByteToWideChar(CP_ACP,0,real_path,-1,NULL,0);
-            if (0 == return_size)
-            {
-                ASSERT("MultiByteToWideChar failure\n");
-                return NULL;
-            }
-
-            return_value = reinterpret_cast<LPWSTR>(InternalMalloc((return_size*sizeof(WCHAR))));
-            if (NULL == return_value)
-            {
-                ERROR("Not enough memory to create full path\n");
-                return NULL;
-            }
-            else
-            {
-                if (!MultiByteToWideChar(CP_ACP, 0, real_path, -1, 
-                                        return_value, return_size))
-                {
-                    ASSERT("MultiByteToWideChar failure\n");
-                    free(return_value);
-                    return_value = NULL;
-                }
-                else
-                {
-                    TRACE("full path to executable is %s\n", real_path.GetString());
-                }
-            }
-
-            return return_value;
+            ASSERT("MultiByteToWideChar failure\n");
+            free(return_value);
+            return_value = NULL;
         }
         else
         {
-            ERROR("found %s in current directory, but it isn't executable!\n",
-                  exe_name);
-        }                                                                   
-    }
-    else
-    {
-        TRACE("last resort failed : executable %s is not in the current "
-              "directory\n",exe_name);
+            TRACE("full path to executable is %s\n", real_path.GetString());
+        }
     }
 
-    ERROR("executable %s not found anywhere!\n", exe_name);
-    return NULL;
+    return return_value;
 #else // !__APPLE__
```

Comment:
I have just noticed that we also have this code for `__APPLE__` here. Could you please remove it and keep just the code you've updated for all platforms?